src/node_jsvmapi.h

```diff
@@ -112,13 +112,18 @@ enum napi_valuetype {
   napi_number,
   napi_string,
   napi_symbol,
+  napi_regexp,
+  napi_date,
+  napi_array,
```

Comment:
Array, Date, and RegExp are not fundamental JavaScript types, they are subtypes of Object. I don't think they belong in this enum, or in `napi_get_type_of_value()`. That is why there is a separate `napi_is_array()` function.